### PR TITLE
Fix for `spawning`/`spawned` events

### PR DESCRIPTION
**Pull request by IRacle1**
Logic before:
Spawning and Spawned events called only for fpc roles(except 0492)
Logic in that pr:
Spawned event called for any situation, spawning called only for fpc roles(with 0492, but with 049 resurrect, for forceclass 0492 spawning event won't fire)
Also, i cant test that💀

### CODE REVIEW

File: Exiled.Events/Handlers/Player.cs [hunk 1]
```diff
@@ -692,8 +694,10 @@ public static class Player
         /// <summary>
         /// Called after a <see cref="API.Features.Player"/> has spawned.
         /// </summary>
-        /// <param name="ev">The <see cref="SpawnedEventArgs"/> instance.</param>
-        public static void OnSpawned(SpawnedEventArgs ev) => Spawned.InvokeSafely(ev);
+        /// <param name="hub">The <see cref="ReferenceHub"/> instance.</param>
+        /// <param name="oldRole">The player's old <see cref="PlayerRoleBase"/> instance.</param>
+        /// <param name="newRole">The player's new <see cref="PlayerRoleBase"/> instance.</param>
+        public static void OnSpawned(ReferenceHub hub, PlayerRoleBase oldRole, PlayerRoleBase newRole) => Spawned.InvokeSafely(new SpawnedEventArgs(hub));
```
Comment (iopietro) [inline Exiled.Events/Handlers/Player.cs:700]: We could add the "OldRole" property to `SpawnedEventArgs`.

File: Exiled.Events/Patches/Events/Player/Spawning.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+                    new(OpCodes.Stloc_S, player.LocalIndex),
+                    new(OpCodes.Brfalse_S, continueLabel),
+
+                    // player
+                    new(OpCodes.Ldloc_S, player.LocalIndex),
+
+                    // position
+                    new(OpCodes.Ldloc_1),
+
+                    // SpawningEventArgs ev = new(player, oldRole, position)
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Spawning.cs:77]:
```suggestion
                    // SpawningEventArgs ev = new(player, position)
```

File: Exiled.Events/Patches/Events/Player/Spawning.cs [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+
+            Label continueLabel = generator.DefineLabel();
+
+            LocalBuilder player = generator.DeclareLocal(typeof(Player));
+            LocalBuilder eventArgs = generator.DeclareLocal(typeof(SpawningEventArgs));
+
+            const int toRemove = 7;
+            int offset = -1;
+
+            int index = newInstructions.FindLastIndex(instruction => instruction.Calls(PropertyGetter(typeof(Component), nameof(Component.transform)))) + offset;
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Spawning.cs:119]:
```suggestion
            const int toRemove = 7;
            int offset = -1;
            int index = newInstructions.FindLastIndex(instruction => instruction.Calls(PropertyGetter(typeof(Component), nameof(Component.transform)))) + offset;
```

File: Exiled.Events/Patches/Events/Player/Spawning.cs [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+                    // player
+                    new(OpCodes.Ldloc_S, player.LocalIndex),
+
+                    // this.ScpRole.FpcModule.Position
+                    new(OpCodes.Ldarg_0),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(ScpStandardSubroutine<Scp049Role>), nameof(ScpStandardSubroutine<Scp049Role>.ScpRole))),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(FpcStandardRoleBase), nameof(FpcStandardRoleBase.FpcModule))),
+                    new(OpCodes.Callvirt, PropertyGetter(typeof(FirstPersonMovementModule), nameof(FirstPersonMovementModule.Position))),
+
+                    // SpawningEventArgs ev = new(player, oldRole, this.ScpRole.FpcModule.Position)
```
Comment (iopietro) [inline Exiled.Events/Patches/Events/Player/Spawning.cs:151]:
```suggestion
                    // SpawningEventArgs ev = new(player, this.ScpRole.FpcModule.Position)
```

**Review by iopietro — CHANGES_REQUESTED** [review]

File: Exiled.Events/Handlers/Player.cs (hunk 1 above)
Comment (IRacle1) [reply Exiled.Events/Handlers/Player.cs:700]: Yea, we can, but we cant add that property to Spawning event, i think it'll break logic

Comment (iopietro) [reply Exiled.Events/Handlers/Player.cs:700]: Sure, add that property only to the `Spawned` event.

Comment (IRacle1) [reply Exiled.Events/Handlers/Player.cs:700]: Uh, ok

**Pull request merged**